src/views/sent_sms_list/StatusFilter.js

```diff
@@ -1,68 +1,48 @@
-import React, { useContext } from 'react'
-import { PropTypes } from '@dhis2/prop-types'
+import React from 'react'
 import { SingleSelectField, SingleSelectOption } from '@dhis2/ui'
+import { useHistory } from 'react-router-dom'
 import i18n from '../../locales'
-import RefetchSms from './RefetchSms'
+import { useQueryParams } from '../../hooks'
+import { createSearchString } from '../../utils'
 import { statusMap } from './translations'
 
-/**
- * Using ALL instead of a more sensible empty string due to a bug in the Select,
- * which means the select doesn't allow empty strings as values.
- * https://github.com/dhis2/ui/issues/245
- */
-
-export const parseStatus = status => {
-    if (status === 'ALL') {
-        return ''
-    }
-
-    return status
-}
-
 // From https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-api/src/main/java/org/hisp/dhis/sms/outbound/OutboundSmsStatus.java
-const statuses = [
+const STATUS_FILTER_OPTIONS = [
+    'ALL',
     'DELIVERED',
     'ERROR',
     'FAILED',
     'OUTBOUND',
     'PENDING',
     'SCHEDULED',
     'SENT',
-]
-const filterOptions = [
-    { label: i18n.t('All'), value: 'ALL' },
-    // Using the translations from statusmap as labels
-    ...statuses.map(status => ({ label: statusMap[status], value: status })),
-]
-
-const StatusFilter = ({ status, setStatus }) => {
-    const { refetchAndClear } = useContext(RefetchSms)
-    const onChange = ({ selected }) => {
-        setStatus(selected)
-        refetchAndClear({
-            status: parseStatus(selected),
-            // Reset to the first page after filtering
-            page: 1,
+].map(status => ({ value: status, label: statusMap[status] }))
+
+const StatusFilter = () => {
+    const { status, pageSize } = useQueryParams()
+    const history = useHistory()
+    const handleStatusChange = ({ selected }) => {
+        history.push({
+            search: createSearchString({
+                status: selected,
+                pageSize,
+                page: 1,
+            }),
         })
     }
 
     return (
         <SingleSelectField
             label={i18n.t('Filter by status')}
             inputWidth="200px"
-            onChange={onChange}
+            onChange={handleStatusChange}
             selected={status}
         >
-            {filterOptions.map(({ label, value }) => (
+            {STATUS_FILTER_OPTIONS.map(({ label, value }) => (
```

Comment:
So this is why I was asking about everyone's preferences on uppercasing in the meeting. Since we don't have an established best practice there this seems fine to me.

Reply:
I made this one uppercase since that's how it was defined in the receive SMS list but yeath we definitely need some kind of linter rule at some point.

Reply:
I agree, going with the established convention in the app sounds like the best approach. (Since we as an organisation haven't decided on a recommendation yet.)